src/main/java/com/solodive/championselect/web/rest/ChampionController.java

```diff
@@ -0,0 +1,58 @@
+package com.solodive.championselect.web.rest;
+
+import com.codahale.metrics.annotation.Timed;
+import com.solodive.championselect.domain.Champion;
+import com.solodive.championselect.service.ChampionService;
+import io.github.jhipster.web.util.ResponseUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * REST controller for managing Champion.
+ */
+@RestController
+@RequestMapping("/api")
+public class ChampionController {
+
+    private final Logger log = LoggerFactory.getLogger(ChampionController.class);
+
+    private static final String ENTITY_NAME = "champion";
+
+    private final ChampionService championService;
+
+    public ChampionController(ChampionService championService) {
+        this.championService = championService;
+    }
+
+    /**
+     * GET  /champions : get all the champions.
+     *
+     * @param eagerload flag to eager load entities from relationships (This is applicable for many-to-many)
+     * @return the ResponseEntity with status 200 (OK) and the list of champions in body
+     */
+    @GetMapping("/champions")
+    @Timed
+    public List<Champion> getAllChampions(@RequestParam(required = false, defaultValue = "false") boolean eagerload) {
+        log.debug("REST request to get all Champions");
+        return championService.findAll();
```

Comment:
logi na poziomie INFO na poczatku i na koncu procesu z uwidocznionymi parametrami requestu oraz rezultatem w responsie (nie cala lista do logu, tylko np liczba zwracanych championow). Do kazdego endpointa w kazdym resourscie

Reply:
Dodatkowo nie obslugujemy tutaj zadnych bledow. Prawdopodobnie jedynie na czym moze pasc, to lacznosc z baza i poleca wtedy jakies randomowe wyjatki. Moze prosty try catch na kazdy exception?

Reply:
Poprawka naniesiona - proszę zweryfikuj.